### ISSUE TITLE
cls() in @classmethod doesn't trigger missing @abstractmethod error

### ISSUE THREAD

**Issue by altendky** [python/mypy#9281]
Bug:

mypy doesn't complain about missing abstract methods when calling `cls()` in a `@classmethod`.  I would expect it to treat the `cls()` and `ForgotTheMethod()` calls the same and produce the missing abstract method error for both.

Yes, I am using `abc` a bit differently.  I don't want a metaclass introduced etc.  I just want a way for mypy to check interfaces for me.

Below was run with `mypy-latest` and `Python 3.8` selected.   mypy 0.782 was listed as latest on PyPI.

https://mypy-play.net/?mypy=latest&python=3.8&gist=fbbbcf4a9c222bef43b133538b219523
```python
import abc
import typing


class AbstractClass:
    @abc.abstractmethod    
    def abstract_method(self) -> int:
        ...


class ForgotTheMethod(AbstractClass):
    @classmethod
    def bare(cls) -> None:
        reveal_type(cls)
        cls()

        reveal_type(ForgotTheMethod)
        ForgotTheMethod()

huh = ForgotTheMethod.bare()
```

```
main.py:14: note: Revealed type is 'Type[main.ForgotTheMethod]'
main.py:17: note: Revealed type is 'def () -> main.ForgotTheMethod'
main.py:18: error: Cannot instantiate abstract class 'ForgotTheMethod' with abstract attribute 'abstract_method'
Found 1 error in 1 file (checked 1 source file)
```

**Comment by gvanrossum** [comment]
IIRC this is intentional. There could be a subclass that *does* define the method.

**Comment by altendky** [comment]
I don't follow.  Why should `cls()` work but `ForgotTheMethod()` not work?  Aren't they both going to create an instance of `ForgotTheMethod`?  My concern is specifically about creating an instance of a class that has abstract methods and it seems like both of the calls qualify.

**Comment by altendky** [comment]
hmm...  i think i get it now.  good thing i guess i was going to move away from classmethods anyways.  :|
